stl/inc/chrono

```diff
@@ -671,7 +695,29 @@ namespace chrono {
         static constexpr bool is_steady = false;
 
         _NODISCARD static time_point now() noexcept { // get current time
-            return time_point(duration(_Xtime_get_ticks()));
+            constexpr long long __std_fs_file_time_epoch_adjustment = 0x19DB1DED53E8000LL; // TRANSITION, ABI
+
+            const duration _File_time{_Xtime_get_ticks() + __std_fs_file_time_epoch_adjustment};
+            const auto _File_seconds = _CHRONO duration_cast<seconds>(_File_time);
+
+            __std_utc_components_1s _Utc;
+            const __std_file_time_to_utc_errc _Ec = __std_file_seconds_to_utc_components(_File_seconds.count(), &_Utc);
+            _STL_INTERNAL_CHECK(_Ec == __std_file_time_to_utc_errc::_Success);
+            if (_Ec == __std_file_time_to_utc_errc::_Invalid_parameter) {
+                return time_point{_File_time - duration{__std_fs_file_time_epoch_adjustment}};
+            }
+
+            using _Days         = _CHRONO duration<int, ratio<86'400>>;
+            using _Int_seconds  = _CHRONO duration<int>;
+            const duration _Ymd = _Days{_Days_from_civil(_Utc._Year, _Utc._Month, _Utc._Day)};
+            const duration _Hms = hours{_Utc._Hour} + minutes{_Utc._Minute} + _Int_seconds{_Utc._Second};
+
+            if (_Utc._Second < 60) {
+                return time_point{_Ymd + _Hms + (_File_time - _File_seconds)};
+            } else {
+                // positive leap second

```

Comment:
The comment "`// positive leap second`" made me wonder how/whether we need to handle potential future _negative_ leap seconds.  I didn't see negative leap seconds referenced in either [Dan's blogpost](https://techcommunity.microsoft.com/t5/networking-blog/leap-seconds-for-the-it-pro-what-you-need-to-know/ba-p/339811) or [Daniel's blogpost](https://techcommunity.microsoft.com/t5/networking-blog/leap-seconds-for-the-appdev-what-you-should-know/ba-p/339813) so I'm wondering if there is any documentation for how a negative leap second would be represented in the Windows API/structs/*?

Reply:
I'm not aware of any documentation explicitly mentioning the handling of negative leap seconds by Windows APIs. My experiments showed that `FileTimeToSystemTime` jumps from 23:59:58.999 to 00:00:00.000 and that `SystemTimeToFileTime` rejects 23:59:59.xxx during a fake negative leap second, whether or not `PROCESS_LEAP_SECOND_INFO_FLAG_ENABLE_SIXTY_SECOND` is set for the process.

Reply:
Code for handling negative leap seconds:
https://github.com/microsoft/STL/blob/7373b982041e46790a715a07e6f716be07ec70b5/stl/src/xtime0.cpp#L102-L109